pymongo/client_session.py

```diff
@@ -330,6 +331,13 @@ def reset(self):
         self.recovery_token = None
         self.attempt = 0
 
+    def __del__(self):
+        if self.sock_mgr:
```

Comment:
Shouldn't this also happen in `reset`?

Reply:
Yes reset does call unpin() which returns the socket right away. On GC we need to use the _close_cursor approach to avoid taking any locks.